src/background/stores/card-selection-store.ts

```diff
@@ -26,6 +26,7 @@ export class CardSelectionStore extends BaseStoreImpl<CardSelectionStoreData> {
     public getDefaultState(): CardSelectionStoreData {
         const defaultValue: CardSelectionStoreData = {
             rules: {},
+            selectedCardCount: 0,
```

Comment:
This information can already be inferred from `rules`, no? Can `get-highlights-from-card-selection-store-data.ts` infer it rather than having to repeat state, eg, by updating lines 22-26 to

```
    const selectedResultUids = getOnlyResultUidsFromSelectedCards(storeData.rules, expandedRuleIds);
    if (selectedResultUids.length > 0) {
        return selectedResultUids;
    }

    return getAllResultUidsFromRuleIdArray(storeData.rules, expandedRuleIds);
```

Reply:
Very nice. That's what I was looking for.